src/test/java/org/swisspush/redisques/slowdown/EnqueueThrottleTest.java

```diff
@@ -0,0 +1,110 @@
+package org.swisspush.redisques.slowdown;
+
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.RunTestOnContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.swisspush.redisques.RedisQues;
+import org.swisspush.redisques.util.QueueConfiguration;
+import org.swisspush.redisques.util.RedisquesAPI;
+import org.swisspush.redisques.util.RedisquesConfiguration;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+import static org.swisspush.redisques.util.RedisquesAPI.*;
+
+@RunWith(VertxUnitRunner.class)
+public class EnqueueThrottleTest {
+
+    private static final String REDISQUES_ADDRESS = "EnqueueThrottleTest-RedisQues";
+    private static final String PROCESSOR_ADDRESS = "EnqueueThrottleTest-Processor";
+    private static final String QUEUE_NAME = "EnqueueThrottleTest-q1";
+
+    @ClassRule
+    public static RunTestOnContext RULE = new RunTestOnContext(new VertxOptions().setBlockedThreadCheckInterval(10_000));
+
+    @Before
+    public void deployRedisQuesAndRegisterProcessor(TestContext testContext) {
+        Async async = testContext.async();
+        Vertx vertx = RULE.vertx();
+
+        RedisquesConfiguration redisquesConfig = RedisquesConfiguration.with()
+                .address(REDISQUES_ADDRESS)
+                .processorAddress(PROCESSOR_ADDRESS)
+                .httpRequestHandlerEnabled(true)
+                .refreshPeriod(2)
+                .checkInterval(10)
+                .queueConfigurations(Arrays.asList(
+                  new QueueConfiguration().withPattern(QUEUE_NAME).withEnqueueDelayMillisPerSize(500).withEnqueueMaxDelayMillis(1700)
+                ))
+                .build();
+
+        vertx.eventBus().consumer(PROCESSOR_ADDRESS, this::processor);
+
+        vertx.deployVerticle(new RedisQues(), new DeploymentOptions().setConfig(redisquesConfig.asJsonObject()), done -> {
+            JsonObject delete = RedisquesAPI.buildDeleteAllQueueItemsOperation(QUEUE_NAME);
+            vertx.eventBus().send(REDISQUES_ADDRESS, delete, deleteDone -> {
+                async.complete();
+            });
+        });
+    }
+
+    private static boolean blockProcessor = true;
+    private static HashSet<String> queueMirror = new HashSet<>();
+
+    private void processor(Message<JsonObject> msg) {
+        String message = msg.body().getString("payload");
+        if (!blockProcessor) {
+            queueMirror.remove(message);
+        }
+        String replyStatus = blockProcessor ? ERROR : OK;
+        System.out.println("Processor received: " + message + " and will reply: " + replyStatus);
```

Comment:
Please the Logger like the other test classes instead of `System.out.println`

Reply:
This is a TEST-Class. Totally new to me that we use a Logger-Infrastructur in tests

Reply:
Actually we do. Take a look at `AbstractTestCase` :-)

Reply:
Ok, can do it. Though I see no win... Can you explain?

Reply:
The win would be _consistency_, but it's not really a show stopper. If you don't want to change it, then don't.